scalafix/rules/src/main/scala/fix/0.8/MigrateV0_8.scala

```diff
@@ -0,0 +1,143 @@
+      case t @ q"def $fn(..$ps): ScioResult = {$body}" =>
+        Patch.addRight(t, ".waitUntilFinish")
+      case t =>
+        Patch.empty
+    }.asPatch
+  }
+}
+
+private class FixScioIO extends SemanticRule("FixScioIO") {
+
+  // Check that the method is a member of an implementation of ScioIO
+  private def isScioIOMember(t: Tree)(implicit doc: SemanticDocument) =
+    t.parent.collect {
+      case p @ Template(_) =>
+        p.inits.exists(_.tpe.symbol == Symbol("com/spotify/scio/io/ScioIO#"))
+    }.getOrElse(false)
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      // fix return type (No Future 🤘)
+      case t @ q"""..$mods def write(
+                      $sName: SCollection[$sTpe],
+                      $psName: $psTpe): $ret = $impl""" if isScioIOMember(t) =>
+        ret.collect {
+          case r @ Type.Apply(Type.Name("Future"), List(tapTpe)) =>
+            Patch.replaceTree(r, tapTpe.syntax)
+        }.asPatch
+    }.asPatch
+  }
+}
+
+private class FixSyntaxImports extends SemanticRule("FixSyntaxImports") {
+  private val imports =
+    scala.collection.mutable.ArrayBuffer.empty[(String, String)]
+
+  // Check that the package is not imported multiple times in the same file
+  def addImport(p: Position, i: Importer) = {
+    val Importer(s) = i
+    val Input.VirtualFile(path, _) = p.input
+
+    val t = (s.toString, path)
+    if(!imports.contains(t)) {
+      imports += t
+      Patch.addGlobalImport(i)
+    } else Patch.empty
+  }
+
+  object JDBC {
+    val fns =
+      List("JdbcScioContext", "JdbcSCollection")
+
+    val `import` = importer"com.spotify.scio.jdbc._"
+  }
+
+  object BQ {
+    val fns =
+      List("toBigQueryScioContext", "toBigQuerySCollection")
+
+    val `import` = importer"com.spotify.scio.bigquery._"
+  }
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    val renameSymbols =
+      Patch.replaceSymbols(
+        "com/spotify/scio/transforms/AsyncLookupDoFn." ->
+          "com/spotify/scio/transforms/BaseAsyncLookupDoFn.",
+        "com.spotify.scio.jdbc.JdbcScioContext." ->
+          "com.spotify.scio.jdbc.syntax.JdbcScioContextOps."
+      )
+
+    doc.tree.collect {
+      case i @ Importee.Name(Name.Indeterminate(n)) if JDBC.fns.contains(n) =>
+        Patch.removeImportee(i) + addImport(i.pos, JDBC.`import`)
+      case i @ Importee.Name(Name.Indeterminate(n)) if BQ.fns.contains(n) =>
+        Patch.removeImportee(i) + addImport(i.pos, BQ.`import`)
+    }.asPatch + renameSymbols
+  }
+}
+
+private class FixWaitForResult extends SemanticRule("FixWaitForResult") {
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    doc.tree.collect {
+      case t @ q"$x.waitForResult()" =>
+        x.symbol.info.get.signature match {
+          // XXX: Hackish wait to check the type of x
+          case ValueSignature(tpe) if tpe.toString.startsWith("WaitableFutureTap") =>
+            Patch.replaceTree(t, x.syntax)
```

Comment:
hm, tbh i think this fix will  be a bit more complex and tricky ... we went from `Future[Tap[T]]` => `ClosedTap[T]` removing `waitForResult` is not enough ... we actually need the result from `sc.close()` ie:

```scala
    val materialized = sc.textFile(input).materialize
    val executionContext = sc.close()
    executionContext.waitUntilDone().tap(materialized).value
```

Reply:
@jto what do you think of removing this rule for now? and get this merged so we can keep iterating on this?

Reply:
I will :)